core/src/v2/actix.rs

```diff
@@ -521,3 +525,120 @@ where
         new.update_definitions(name, map);
     }
 }
+
+macro_rules! json_with_status {
+    ($name:ident => $status:expr) => {
+        pub struct $name<T: Serialize + Apiv2Schema>(pub T);
+
+        impl<T> fmt::Debug for $name<T>
+        where
+            T: fmt::Debug + Serialize + Apiv2Schema,
+        {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                let status: StatusCode = $status;
+                let status_str = status.canonical_reason().unwrap_or(status.as_str());
+                write!(f, "{} Json: {:?}", status_str, self.0)
+            }
+        }
+
+        impl<T> fmt::Display for $name<T>
+        where
+            T: fmt::Display + Serialize + Apiv2Schema,
+        {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                fmt::Display::fmt(&self.0, f)
+            }
+        }
+
+        impl<T> Responder for $name<T>
+        where
+            T: Serialize + Apiv2Schema,
+        {
+            type Error = Error;
+            type Future = Ready<Result<HttpResponse, Error>>;
+
+            fn respond_to(self, _: &HttpRequest) -> Self::Future {
+                let status: StatusCode = $status;
+                let body = match serde_json::to_string(&self.0) {
+                    Ok(body) => body,
+                    Err(e) => return ready(Err(e.into())),
+                };
+
+                ready(Ok(HttpResponse::build(status)
+                    .content_type("application/json")
+                    .body(body)))
+            }
+        }
+
+        impl<T> Apiv2Schema for $name<T>
+        where
+            T: Serialize + Apiv2Schema,
+        {
+            const NAME: Option<&'static str> = T::NAME;
+
+            fn raw_schema() -> DefaultSchemaRaw {
+                T::raw_schema()
+            }
+        }
+
+        impl<T> OperationModifier for $name<T>
+        where
+            T: Serialize + Apiv2Schema,
+        {
+            fn update_response(op: &mut DefaultOperationRaw) {
+                let status: StatusCode = $status;
+                op.responses.insert(
+                    status.as_str().into(),
+                    Either::Right(Response {
+                        description: status.canonical_reason().map(ToString::to_string),
+                        schema: Some({
+                            let mut def = T::schema_with_ref();
+                            def.retain_ref();
+                            def
+                        }),
+                        ..Default::default()
+                    }),
+                );
+            }
+        }
+    };
+}
+
+json_with_status!(CreatedJson => StatusCode::CREATED);
+json_with_status!(AcceptedJson => StatusCode::ACCEPTED);
+
+#[derive(Debug)]
+pub struct NoContent;
```

Comment:
Do we have to implement this manually? I understand it's because `NoContent` is not supposed to return anything, but still, we can use `json_with_status!` for `NoContent` too.

Reply:
I'm for making this DRY, but I think if the behaviour of `NoContent` was serialising `()` as json it would respond with `null`. So i'd need to make the approach more generic and make it apply to any responder rather than reimplementing Json. I like it but I'm not sure how to do it yet